callout/src/test/java/com/google/apigee/ProtoMessageBuilders/ExecutionContextProtoMessageBuilderTest.java

```diff
@@ -0,0 +1,270 @@
+    doReturn(FAULT_SUB_CATEGORY).when(mockFault).getSubCategory();
+
+    executionContext.setRequestFlow(false);
+    executionContext.setErrorFlow(false);
+    executionContext.clearFaults();
+  }
+
+  @Test
+  public void testBuildExecutionContextProtoDefault() {
+    Execute.ExecutionContext proto =
+        ExecutionContextProtoMessageBuilder.buildExecutionContextProto(executionContext);
+
+    assertEquals(0, proto.getFaultsCount());
+    assertEquals(Execute.ExecutionContext.FlowType.UNKNOWN_FLOW_TYPE, proto.getFlowType());
+  }
+
+  @Test
+  public void testBuildExecutionContextProtoFlowTypeRequest() {
+    executionContext.setRequestFlow(true);
+    Execute.ExecutionContext proto =
+        ExecutionContextProtoMessageBuilder.buildExecutionContextProto(executionContext);
+
+    assertEquals(0, proto.getFaultsCount());
+    assertEquals(Execute.ExecutionContext.FlowType.REQUEST, proto.getFlowType());
+  }
+
+  @Test
+  public void testBuildExecutionContextProtoFlowTypeError() {
+    executionContext.setErrorFlow(true);
+    Execute.ExecutionContext proto =
+        ExecutionContextProtoMessageBuilder.buildExecutionContextProto(executionContext);
+
+    assertEquals(0, proto.getFaultsCount());
+    assertEquals(Execute.ExecutionContext.FlowType.ERROR, proto.getFlowType());
+  }
+
+  @Test
+  public void testBuildExecutionContextProtoFlowTypeErrorAndRequest() {
+    executionContext.setErrorFlow(true);
+    executionContext.setRequestFlow(true);
+    Execute.ExecutionContext proto =
+        ExecutionContextProtoMessageBuilder.buildExecutionContextProto(executionContext);
+
+    assertEquals(0, proto.getFaultsCount());
+    assertEquals(Execute.ExecutionContext.FlowType.ERROR, proto.getFlowType());
+  }
+
+  @Test
+  public void testBuildExecutionContextProtoOneFault() throws Exception {
+    executionContext.addFault(mockFault);
+    Execute.ExecutionContext proto =
+        ExecutionContextProtoMessageBuilder.buildExecutionContextProto(executionContext);
+
+    assertEquals(1, proto.getFaultsCount());
+    assertEquals(
+        Execute.ExecutionContext.Fault.Category.MESSAGING, proto.getFaults(0).getCategory());
+    assertEquals(FAULT_SUB_CATEGORY, proto.getFaults(0).getSubCategory());
+    assertEquals(FAULT_NAME, proto.getFaults(0).getName());
+    assertEquals(FAULT_REASON, proto.getFaults(0).getReason());
+    assertEquals(0, proto.getFaults(0).getAttributesCount());
+    assertEquals(Execute.ExecutionContext.FlowType.UNKNOWN_FLOW_TYPE, proto.getFlowType());
+  }
+
+  @Test
+  public void testBuildExecutionContextProtoWithMultipleFaults() {
+    executionContext.addFault(mockFault);
+    executionContext.addFault(mockFault);
+    doReturn(Fault.Category.Messaging).when(mockFault).getCategory();
+    Execute.ExecutionContext proto =
+        ExecutionContextProtoMessageBuilder.buildExecutionContextProto(executionContext);
+
+    assertEquals(2, proto.getFaultsCount());
+    assertEquals(
```

Comment:
nit: could you try to reformat this file with `google-java-format`? I think lines 126 and 132 are the same but they are line separated differently